### PR TITLE
Issue 78

### PR DESCRIPTION
**Pull request by bnandaku**
I can successfully populate a page for get-html if the hosted site is
on HTTP instead of HTTPS.

This is not for merge, but rather for comments.

### CODE REVIEW

**Comment by bnandaku** [comment]
@san983 The DOM morphs the information, so I have a lot of trouble trying to recreate the button. Aside from actually extracting the information. Shall I look around for an alternative method for this then ?

**Comment by bnandaku** [comment]
@san983 I was able to do the requested changes. Please take a look and give me an opinion.

File: public/javascripts/click2vox-1.5.1.js [hunk 1]
```diff
@@ -603,7 +603,7 @@ var check1Ready = (function() {
     } else {
       var buttonData = document.querySelector('.voxButton');
       // console.log(buttonData.dataset);
-      open('POST', infoVoxbone.server_url + '/click2vox/portal-widget/get-html', buttonData.dataset, 'newwin');
+      open('POST', infoVoxbone.server_url + '/portal-widget/get-html', buttonData.dataset);
```
Comment (san983) [inline public/javascripts/click2vox-1.5.1.js]: 👍

File: views/layout.jade [hunk 2]
```diff
@@ -17,6 +17,7 @@ html(lang='en')
     link(href="/click2vox/stylesheets/root.css" rel="stylesheet")
     link(href="//cdnjs.cloudflare.com/ajax/libs/raty/2.7.0/jquery.raty.min.css" rel="stylesheet")
     link(href="//cdn.jsdelivr.net/qtip2/2.2.1/jquery.qtip.min.css" rel="stylesheet")
+    link(rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/github-fork-ribbon-css/0.2.0/gh-fork-ribbon.min.css")
```
Comment (san983) [inline views/layout.jade]: Remove `https:` just keep `//`

File: public/html/widget_popup.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<div data-text="Hello%20World" data-did="16092975382" data-incompatible_browser_configuration="hide_widget" data-server_url="https://portal-widget-generator--pr-98.herokuapp.com" data-use_default_button_css="true" data-div_css_class_name="style-a" data-basic_auth="true" data-voxbone_webrtc_username="san983" data-voxbone_webrtc_password="Click456!" class="voxButton"></div><script src="https://portal-widget-generator--pr-98.herokuapp.com/click2vox.js"></script>
```
Comment (emilianop11) [inline public/html/widget_popup.html]: Whats this file? Why is the server url hardcoded?

File: public/javascripts/click2vox-1.5.0.js [hunk 4]
```diff
@@ -46,6 +46,7 @@ var check0Ready = (function() {
 
   if (typeof voxbone === 'undefined')
     loadScript("//cdn.voxbone.com/voxbone/voxbone-2.1.min.js", check1Ready);
+
```
Comment (emilianop11) [inline public/javascripts/click2vox-1.5.0.js]: Remove extra line please

File: public/javascripts/widget-0.1.js [hunk 5]
```diff
@@ -275,6 +275,10 @@ $(document).ready(function () {
           else if (e.localVolume <= 0.01) postMessage("setMicVolume0","*")
     }
 
+    voxbone.WebRTC.customEventHandler.remoteMediaVolume = function(e) {
+        console.log(e.remoteVolume);
```
Comment (emilianop11) [inline public/javascripts/widget-0.1.js]: Is this needed?

File: public/javascripts/widget-0.2.js [hunk 6]
```diff
@@ -278,6 +278,9 @@ $(document).ready(function () {
     else if (e.localVolume > 0.01) postMessage("setMicVolume1","*")
     else if (e.localVolume <= 0.01) postMessage("setMicVolume0","*")
   }
+  voxbone.WebRTC.customEventHandler.remoteMediaVolume = function(e) {
+      console.log(e.remoteVolume);
```
Comment (emilianop11) [inline public/javascripts/widget-0.2.js]: Same here

File: public/javascripts/widget-0.3.js [hunk 7]
```diff
@@ -367,6 +367,9 @@ function loadWidget() {
     else if (e.localVolume > 0.01) postMessage("setMicVolume1","*")
     else if (e.localVolume <= 0.01) postMessage("setMicVolume0","*")
   }
+  voxbone.WebRTC.customEventHandler.remoteMediaVolume = function(e) {
+      console.log(e.remoteVolume);
```
Comment (emilianop11) [inline public/javascripts/widget-0.3.js]: Same here

File: public/javascripts/widget-0.4.js [hunk 8]
```diff
@@ -355,6 +355,9 @@ function loadWidget() {
     else if (e.localVolume > 0.01) postMessage("setMicVolume1","*")
     else if (e.localVolume <= 0.01) postMessage("setMicVolume0","*")
   }
+  voxbone.WebRTC.customEventHandler.remoteMediaVolume = function(e) {
+      console.log(e.remoteVolume);
```
Comment (emilianop11) [inline public/javascripts/widget-0.4.js]: and here

File: public/stylesheets/less/ngToast.less [hunk 9]
```diff
@@ -12,3 +11,64 @@
   background-color: #f2dede;
   border-color: #ebccd1;
 }
+
```
Comment (emilianop11) [inline public/stylesheets/less/ngToast.less]: Please check indentation and brackets position on the whole file

File: routes/widget.js [hunk 10]
```diff
@@ -88,6 +88,20 @@ router.post('/portal-widget/get-code', function(req, res, next) {
   }
 });
 
+router.post('/portal-widget/get-html', function(req, res, next){
+  var result = {};
+  var widget = req.parameters
+  try {
+    result.widget_code = utils.widgetSecureDivhtml(widget, widget.did);
+    res.render('widget/widget_html', result);
+
```
Comment (emilianop11) [inline routes/widget.js]: Remove extra line please

File: routes/utils.js [hunk 11]
```diff
@@ -50,5 +50,19 @@ module.exports = {
     };
 
     return jade.renderFile('./views/voxbone_widget_div.jade', params);
+  },
+  widgetSecureDivhtml: function (widget, did) {
+    var jade = require('jade');
```
Comment (emilianop11) [inline routes/utils.js]: We should be using pug not jade

File: views/layout.jade [hunk 12]
```diff
@@ -30,16 +31,15 @@ html(lang='en')
     block extra_head
 
   body
-    toast
-    a(href="https://github.com/voxbone-workshop/portal-widget")
-      img(style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/a6677b08c955af8400f44c6298f40e7d19cc5b2d/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f677261795f3664366436642e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_gray_6d6d6d.png")
-    .navbar-container
+  toast
```
Comment (emilianop11) [inline views/layout.jade]: The indentation here is wrong

File: views/layout.jade [hunk 13]
```diff
@@ -30,16 +31,15 @@ html(lang='en')
     block extra_head
 
   body
-    toast
-    a(href="https://github.com/voxbone-workshop/portal-widget")
-      img(style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/a6677b08c955af8400f44c6298f40e7d19cc5b2d/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f677261795f3664366436642e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_gray_6d6d6d.png")
-    .navbar-container
+  a(class="github-fork-ribbon" href="https://github.com/voxbone-workshop/portal-widget")
```
Comment (emilianop11) [inline views/layout.jade]: Wrong indentation

File: public/stylesheets/less/ngToast.less [hunk 14]
```diff
@@ -12,3 +11,64 @@
...
+/*!
+ * ngToast v2.0.0 (http://tameraydin.github.io/ngToast)
+ * Copyright 2016 Tamer Aydin (http://tamerayd.in)
+ * Licensed under MIT (http://tameraydin.mit-license.org/)
+ */
+
+.ng-toast {
```
Comment (emilianop11) [inline public/stylesheets/less/ngToast.less]: File has a lot of indentation problems

**Comment by san983** [comment]
Please @bnandaku, create a New Branch with a new PR including just the commits / file changes for this issue #78. BE aware of the indentation and if there is some other issues you're addressing open a new branch/PR for it. Thanks

**Pull request closed without merging**